File: sample_objc/ExelbidSample/Views/EBFanNativeAdView.m
//
//  EBFanNativeAdView.m
//
//  Created by Jaeuk Jeong on 2/29/24.
//

#import "EBFanNativeAdView.h"

@interface EBFanNativeAdView ()

@end

@implementation EBFanNativeAdView

- (void)awakeFromNib {
    [super awakeFromNib];
}

- (id)initWithNativeAd:(FBNativeAd *)nativeAd frame:(CGRect)frame
{
    self = [super initWithFrame:frame];
    if (self) {
        _adTitleLabel.text = nativeAd.advertiserName;
        _adBodyLabel.text = nativeAd.bodyText;
        _adSocialContextLabel.text = nativeAd.socialContext;
        _sponsoredLabel.text = nativeAd.sponsoredTranslation;
        [_adCallToActionButton setTitle:nativeAd.callToAction forState:UIControlStateNormal];
        _adOptionsView.nativeAd = nativeAd;
    }
    return self;
}

@end

File: sample/ExelbidSample/Views/EBFanNativeAdView.swift
//
//  EBFanNativeAdView.swift
//  ExelbidSample
//
//  Created by Jaeuk Jeong on 3/19/24.
//

import Foundation
import UIKit
import FBAudienceNetwork

class EBFanNativeAdView : UIView {
    var adView: UIView!

    @IBOutlet var adIconImageView: FBMediaView!
    @IBOutlet var adCoverMediaView: FBMediaView!
    @IBOutlet var adTitleLabel: UILabel!
    @IBOutlet var adBodyLabel: UILabel!
    @IBOutlet var adCallToActionButton: UIButton!
    @IBOutlet var adSocialContextLabel: UILabel!
    @IBOutlet var sponsoredLabel: UILabel!
    @IBOutlet var adChoicesView: FBAdChoicesView!
    @IBOutlet var adOptionsView: FBAdOptionsView!
    
    override class func awakeFromNib() {
        super.awakeFromNib()
    }
    
}
